ecs_model_deployer/src/index.ts

```diff
@@ -88,56 +80,34 @@ export const handler = async (event: any) => {
 
 
     const stackName = `${config.deploymentName}-${modelConfig.modelId}`;
-    if ( event.action === ACTION_DEPLOY ) {
-        const deploy_promise: Promise<ChildProcess | undefined> = new Promise( (resolve) => {
-            const cp = spawn('./node_modules/aws-cdk/bin/cdk', ['deploy', stackName, '-o', '/tmp/cdk.out']);
-
-            cp.on('close', (code) => {
-                console.log(`cdk deploy exited early, code ${code}`);
-                resolve(cp);
-            });
-
-            cp.stdout.on('data', (data) => {
-                console.log(`Got data: ${data}`);
-            });
-
-            cp.stderr.on('data', (data) => {
-                console.log(`Got err data: ${data}`);
-            });
-
-            setTimeout(() => {
-                console.log('180 second timeout');
-                resolve(undefined);
-            }, 180 * 1000);
+    const deploy_promise: Promise<ChildProcess | undefined> = new Promise( (resolve) => {
+        const cp = spawn('./node_modules/aws-cdk/bin/cdk', ['deploy', stackName, '-o', '/tmp/cdk.out']);
+
+        cp.on('close', (code) => {
+            console.log(`cdk deploy exited early, code ${code}`);
+            resolve(cp);
         });
 
-        const cp = await deploy_promise;
-        if ( cp ) {
-            if ( cp.exitCode !== 0 ) {
-                throw new Error('Stack failed to deploy');
-            }
-        }
-    } else if ( event.action === ACTION_DESTROY ) {
-        const deploy_promise: Promise<Number> = new Promise( (resolve) => {
-            const cp = spawn('./node_modules/aws-cdk/bin/cdk', ['destroy', '-f', stackName, '-o', '/tmp/cdk.out']);
-
-            cp.on('close', (code) => {
-                resolve(code ?? -1);
-            });
-
-            setTimeout(() => {
-                console.log('60 second timeout');
-                resolve(0);
-            }, 180 * 1000);
+        cp.stdout.on('data', (data) => {
+            console.log(`Got data: ${data}`);
         });
 
-        const exitCode = await deploy_promise;
-        stdout = String(ret.output[1]);
-        stderr = String(ret.output[2]);
-        if ( exitCode !== 0 ) {
-            console.log(`cdk destroy failed with stdout: ${stdout}, stderr: ${stderr}`);
-            throw new Error('Stack failed to destroy');
+        cp.stderr.on('data', (data) => {
+            console.log(`Got err data: ${data}`);
+        });
+
+        setTimeout(() => {
+            console.log('180 second timeout');
+            resolve(undefined);
+        }, 180 * 1000);
+    });
```

Comment:
What's going on in this block? 3 minute wait? Are we expecting this lambda to stay running the entire duration of the cdk update?

Reply:
We need to wait long enough to be confident that cdk has under the hood called `aws cloudformation create-stack`. If we exit before this time, the deployment will not happen. We are essentially timing a "control-c" to `cdk deploy` to be late enough that it doesn't matter. I wanted to poll on cloudformation describestacks but unfortunately @aws-sdk/client-cloudformation took us over 250MB...

Reply:
but if CDK fails within the 3 minutes (like a synth fail) we will still capture that error.

Reply:
got it- so the 3 minutes is really just a brute-force "really make sure that the stack has started by this point" then? is it possible that we may not see that in 3 minutes? would it maybe make sense for us to make another wait+poll loop to see if the stack exists (since we know the name of the stack before starting the update)?

Reply:
the problem is we can't do that inside of the ecs_model_deployer lambda b/c of the 250MB limit. So we'd have to have it somewhere else in the state machine, which we essentially already do. The first time we hit the stack poller and it doesn't see that stack as CREATE_COMPLETE, CREATE_IN_PROGRESS, UPDATE_COMPLETE, UPDATE_IN_PROGRESS it will throw and abort the state machine with failure

Reply:
we're not locked into the current architecture- if we need to add one more polling loop in the state machine we can totally do that. I'm very ok with offloading that logic into another loop (outside of the large lambda func) so that we can track statuses appropriately and without missing an update.

i see it as:
loop 1: start docker copy -> poll docker loop
loop 2 (potential new loop): start cdk -> poll for stack creation started loop (don't even have to check status, just that the stack exists)
loop 3: check stack _success_ loop